test/runner/reporter.js

```diff
@@ -3,6 +3,13 @@ import { getBrowserString } from "./lib/getBrowserString.js";
 import { prettyMs } from "./lib/prettyMs.js";
 import * as Diff from "diff";
 
+function quoteWrap( value ) {
+	if ( typeof value === "string" ) {
+		return `"${ value }"`;
```

Comment:
We should also escape inner quotes in such a case.

Reply:
I've tested if the value has quotes and they are already escaped by the template literal.

Reply:
What do you mean? `value` is just the value from the test, I assume. If it contains double quotes and you print the string and the printing utility indicates the string by double quotes then yes, it will escape them. But the string doesn't include any escaping.

In other words, if `value === 'foo "bar" baz'`, then this function will return `'"foo "bar" baz"'` which is not what we want.

Reply:
I'm not sure I understand. I *do* want quotes to be left alone and have the whole string still be wrapped by quotes if the value is a string.

Reply:
`'"foo "bar" baz"'` <- this is exactly what I want, but I think it would actually print as `'"foo \"bar\" baz"'`, which is fine with me. I want it clear the value is a string and the other value is not (we already know they are not both strings) and that the string in fact contains double quotes.

Reply:
I disagree `"foo "bar" baz"` is what we want. This looks nonsensical and is a JS syntax error. If we indicate a string by outer quotes, inner quotes should be escaped. Right now, for:
```js
assert.equal( 42, "foo \"bar\" baz", "message" );
```
we get:
```
expected: "foo \"bar\" baz"
actual: 42
"foo "bar" baz"42
```

The diff looks extremely confusing to me because it looks like we have two strings in there with a weird `bar` between them. I'd expect to see:
```
expected: "foo \"bar\" baz"
actual: 42
"foo \"bar\" baz"42
```
Or just skip the diff in this case. We get the biggest benefits from the diff from matching actual & expected types. Then we can skip the manual quoting & using `diffWords` completely.

A slightly unrelated remark - we are not printing any data for mismatched symbols. The object we get in the reporter is missing `actual` & `expected` properties there, perhaps because symbols by default are skipped during `JSON.stringify`. But perhaps it's fine here, it's not like we'll have many symbol-related tests...

Reply:
ok, I thought I had seen it print with escape backslashes already, but that was wrong. I've finally come around and will escape the double quotes.